runner/src/bootstrap.rs

```diff
@@ -52,30 +54,34 @@ pub async fn bootstrap(opts: Opts) -> Result<()> {
 }
 
 #[tracing::instrument(skip(peers), fields(peers.len = peers.len()))]
-async fn ring(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
-    for peer_info in peers.values() {
+async fn ring(n: usize, peers: &BTreeMap<PeerIdx, Peer>) -> Result<()> {
+    for peer in peers.values() {
         // Connect to each peer in a ring.
-        // Do not attempt to connect to more peers than exist.
-        for i in 0..min(n, peers.len() - 1) {
-            let idx = peer_info.index as usize;
-            let other = (idx + i + 1) % peers.len();
-            debug!(%idx, %other, "ring peer connection");
-            if let Some(other_info) = peers.get(&other) {
-                if let Err(err) = connect_peers(peer_info, other_info).await {
-                    error!(
-                        ?peer_info.index,
-                        ?other_info.index,
-                        ?err,
-                        "failed to bootstrap ring peer"
-                    );
-                }
+        for (_, other) in peers
+            .iter()
+            .cycle()
```

Comment:
this repeats the iter and is similar to taking mod before?

Reply:
Yes, this makes the iteration start back at the beginning when it gets to the end. But now we don't have to do index math like we were before, and since indexes can be negative its a good change.